### PR TITLE
Change deprecation.elasticsearch keyword to elasticsearch.deprecation

### LINKED ISSUE #83251: Inconsistent value of event.dataset in ES deprecation logs
### Elasticsearch Version

8.0

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

macos

### Problem Description

 event.dataset values in ES logs are `elasticsearch.server`, `elasticsearch.index_search_slowlog`, `elasticsearch.index_indexing_slowlog`
but for deprecation logs it is `deprecation.elasticsearch`

This probably originates from 7.x where the deprecation log had deprecation.elasticsearch type field, but other logs had simply server, index_search_slowlog and index_indexing_slowlog

### Steps to Reproduce

generate deprecation logs.
A sample:
```
{
  "@timestamp": "2022-01-27T11:48:45.809Z",
  "log.level": "CRITICAL",
  "data_stream.dataset": "deprecation.elasticsearch",
  "data_stream.namespace": "default",
  "data_stream.type": "logs",
  "elasticsearch.elastic_product_origin": "",
  "elasticsearch.event.category": "compatible_api",
  "elasticsearch.http.request.x_opaque_id": "v7app",
  "event.code": "create_index_with_types",
  "message": "[types removal] Using include_type_name in create index requests is deprecated. The parameter will be removed in the next major version.",
  "ecs.version": "1.2.0",
  "service.name": "ES_ECS",
  "event.dataset": "deprecation.elasticsearch",
  "process.thread.name": "elasticsearch[runTask-0][transport_worker][T#8]",
  "log.logger": "org.elasticsearch.deprecation.rest.action.admin.indices.RestCreateIndexAction",
  "trace.id": "0af7651916cd43dd8448eb211c80319c",
  "elasticsearch.cluster.uuid": "5alW33KLT16Lp1SevDqDSQ",
  "elasticsearch.node.id": "tVLnAGLgQum5ca6z50aqbw",
  "elasticsearch.node.name": "runTask-0",
  "elasticsearch.cluster.name": "runTask"
}

```

beats processor that is overriding this already
https://github.com/elastic/beats/pull/30018/files#r794578451

original dicussion on the PR that introduced the change
https://github.com/elastic/elasticsearch/pull/68737

we should also discuss `datastream.dataset` field. `datastream.dataset` is only set in deprecation logs

### PR DESCRIPTION
**Pull request by JVerwolf**
This PR changes the values of the `"data_stream.dataset"` and `"event.dataset"` fields to be `"elasticsearch.deprecation"` instead of `"deprecation.elasticsearch"`.

Closes https://github.com/elastic/elasticsearch/issues/83251

### CODE REVIEW

File: distribution/src/config/log4j2.properties [hunk 1]
```diff
@@ -63,7 +63,7 @@ appender.deprecation_rolling.name = deprecation_rolling
 appender.deprecation_rolling.fileName = ${sys:es.logs.base_path}${sys:file.separator}${sys:es.logs.cluster_name}_deprecation.json
 appender.deprecation_rolling.layout.type = ECSJsonLayout
 # Intentionally follows a different pattern to above
```
Comment (rjernst) [inline distribution/src/config/log4j2.properties:65]: This comment can be removed

File: docs/changelog/117933.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pr: 117933
+summary: Change `deprecation.elasticsearch` keyword to `elasticsearch.deprecation`
+area: Infra/Logging
+type: bug
```
Comment (rjernst) [inline docs/changelog/117933.yaml:4]: We should note this as a breaking change, since some users may be relying on the ECS dataset.

**Review by rjernst — COMMENTED** [review]
Looks good. We just need to add some breaking change explanation.

File: docs/changelog/117933.yaml (hunk 2 above)
Comment (JVerwolf) [reply docs/changelog/117933.yaml]: Thanks @rjernst - I've updated it now, WDYT?

**Review by rjernst — APPROVED** [review]
LGTM

File: docs/changelog/117933.yaml (hunk 2 above)
Comment (masseyke) [reply docs/changelog/117933.yaml:4]: It breaks more than just search, right? I've seen this in the logs of a cluster that someone had upgraded across versions:
```
[instance-0000000005] Bulk write of deprecation logs encountered some failures: [[4VK3j5UBamJ0EqvgUJmJ org.elasticsearch.index.mapper.DocumentParsingException: [1:87] failed to parse field [data_stream.dataset] of type [constant_keyword] in document with id '4VK3j5UBamJ0EqvgUJmJ'. Preview of field's value: 'elasticsearch.deprecation']]
```
So we're losing writes to the deprecation log until a rollover happens on the data stream.

**Comment by leemthompo** [comment]
@JVerwolf is this PR relevant to the serverless changelog? [FYI this question is based on [9.0 breaking changes](https://staging-website.elastic.co/docs/release-notes/elasticsearch/breaking-changes)]

**Comment by JVerwolf** [comment]
@leemthompo Yup, I believe so. What am I missing here? Is there something else that needs to be included?

**Pull request merged**